hs_geo_raster_resource/serialization.py

```diff
@@ -0,0 +1,267 @@
+            # Get variableUnit
+            varunit_lit = self._rmeta_graph.value(o, hsterms.variableUnit)
+            if varunit_lit is None:
+                msg = "variableUnit for BandInformation was not found for resource {0}".format(self.root_uri)
+                raise GenericResourceMeta.ResourceMetaException(msg)
+            self.band_info.variableUnit = str(varunit_lit)
+            # Get method
+            method_lit = self._rmeta_graph.value(o, hsterms.method)
+            if method_lit is not None:
+                self.band_info.method = str(method_lit)
+            # Get comment
+            comment_lit = self._rmeta_graph.value(o, hsterms.comment)
+            if comment_lit is not None:
+                self.band_info.comment = str(comment_lit)
+            print("\t\t{0}".format(self.band_info))
+
+        # Get spatialReference
+        for s, p, o in self._rmeta_graph.triples((None, hsterms.spatialReference, None)):
+            print("Subject: {0}\npred: {1}\nobj: {2}\n".format(s, p, o))
+            spat_ref_lit = self._rmeta_graph.value(o, rdflib.namespace.RDF.value)
+            if spat_ref_lit is None:
+                msg = "Spatial reference value not found for {0}.".format(o)
+                raise GenericResourceMeta.ResourceMetaException(msg)
+            self.spatial_reference = RasterResourceMeta.SpatialReference(str(spat_ref_lit))
+            print("\t\t{0}".format(self.spatial_reference))
+
+    def write_metadata_to_resource(self, resource):
+        """
+        Write metadata to resource
+
+        :param resource: RasterResource instance
+        """
+        super(RasterResourceMeta, self).write_metadata_to_resource(resource)
+
+        # import sys
+        # sys.path.append("/home/docker/pycharm-debug")
+        # import pydevd
+        # pydevd.settrace('selimnairb.dyndns.org', port=21000, suspend=False)
+
+        if self.cell_info:
+            cell_info = resource.metadata._cell_information.all()[0]
+            kwargs = {'name': self.cell_info.name,
+                      'rows': self.cell_info.rows,
+                      'columns': self.cell_info.columns,
+                      'cellSizeXValue': self.cell_info.cellSizeXValue,
+                      'cellSizeYValue': self.cell_info.cellSizeYValue,
+                      'cellDataType': self.cell_info.cellDataType,
+                      'content_object': resource}
+            CellInformation.update(cell_info.id, **kwargs)
+            resource.save()
+        if self.band_info:
+            resource.metadata._band_information.all().delete()
+            kwargs = {'name': self.band_info.name,
+                      'variableName': self.band_info.variableName,
+                      'variableUnit': self.band_info.variableUnit,
+                      'method': self.band_info.method,
+                      'comment': self.band_info.comment,
+                      'content_object': resource}
+            BandInformation.create(**kwargs)
```

Comment:
@selimnairb The metadata element for specific resource type is also created in the same way as other core metadata elements as shown here: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/hydroshare/resource.py#L409-L413. Take a look at hs_geo_raster_resource/receivers.py raster_pre_create_resource_trigger() to see how extended metadata elements are appended to metadata dictionary, which is further passed to metadata input parameter of create_resource() function which will create extended metadata elements for specific resource type when the resource is created from create_resource() call.